po/zh-Hans.po

```diff
@@ -2269,12 +2278,17 @@ msgid ""
 "* In fact, some will consider infinitely populating a collection a memory\n"
 "  leak and Rust does not protect from those."
 msgstr ""
+"* 可以使用 [`Box::leak`] 来泄漏一个指针。该方法可以用于得到在运行时决定大小和初始化的静态变量\n"
+"* 可以使用 [`std::mem::forget`] 来让编译器“忘记”一个值（即其析构函数不会被执行）。\n"
+"* 可以使用 `Rc` 或 `Arc` 意外创建一个 [循环引用]。\n"
```

Comment:
Here's a link of "reference cycle", if change it to Chinese the link will become a raw text:
<img width="913" alt="image" src="https://github.com/google/comprehensive-rust/assets/42133316/345cc22d-61ee-4ccd-9cbd-409f50a72d79">
<img width="801" alt="image" src="https://github.com/google/comprehensive-rust/assets/42133316/e58049cf-d20a-4a78-a908-672ea6f40c41">
```suggestion
"* 可以使用 `Rc` 或 `Arc` 意外创建一个循环引用（[reference cycle]）。\n"
```

Reply:
I wonder is there any other solution other than keep the original text to be able to use these links? I searched for "reference cycle" but cannot find that in other places in the po file.

Reply:
@sylxjtu, it's tricky! I wrote a bit about the problem in our [style guide](https://github.com/google/comprehensive-rust/blob/main/STYLE.md#translations). In short, we only extract the text part from the Markdown files, so the PO files don't have the link _definition_. They only have the link _usage_, so you have to be careful to align the use with the definition.

This is clearly something that could be improved in https://github.com/google/mdbook-i18n-helpers/, but I'm not sure exactly how.

Reply:
@mgeisler Oh I got it, there are some hidden line not shown in markdown preview mode which represents the link, and we need to add it to our po file like below: 
<img width="1253" alt="image" src="https://github.com/google/comprehensive-rust/assets/42133316/5a92abfe-aa18-463f-9d66-131790289b4b">
But I found it doesn't work in zh-Hans.po, pt-BR.po and es.po like this: 
<img width="755" alt="image" src="https://github.com/google/comprehensive-rust/assets/42133316/7ea72497-5f18-487a-9cb2-8a4e82216afa">
I wonder how to fix this.

Reply:
@wnghl, yes, you got it! The translation system is really a very indirect way of letting you construct a Markdown document. This means that you can inject arbitrary Markdown in your translations (such as `**foo**` and `[foo](http://example.com)`) and it also means that you have to keep the Markdown rules vaguely in mind.

Here the problem is in the pt-BR translation: they should not have translated the reference link (@rastringer and @hugojacob, FYI) since it refers to a link definition which is outside the PO file.

I'll send a PR for this next since it's easy to fix when you know how (and easy to fix even without speaking Portuguese) :smile:

Reply:
I originally used the short

```markdown
foo [bar] baz

bar: http://example.net/
```

style links since I think they look great in the English source text. However, because of how they make translations harder, I'm thinking of switching them all to the inline style:

```markdown
foo [bar](http://example.net/) baz
```

What would make it much harder for the links to go out of sync in the translation and it would also allow the translator to replace the link with a different one, e.g., to the local Wikipedia page or so.